instrumentation/servlet/servlet-common/bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/servlet/ExperimentalSnippetHolder.java

```diff
@@ -5,16 +5,18 @@
 
 package io.opentelemetry.javaagent.bootstrap.servlet;
 
+import java.util.concurrent.atomic.AtomicReference;
+
 public class ExperimentalSnippetHolder {
 
-  private static volatile String snippet = "";
+  private static final AtomicReference<String> snippet = new AtomicReference<>(System.getProperty("otel.experimental.javascript-snippet", ""));
 
-  public static void setSnippet(String snippet) {
-    ExperimentalSnippetHolder.snippet = snippet;
+  public static void setSnippet(String newValue) {
+    snippet.compareAndSet("", newValue);
```

Comment:
I'd like to just confirm we're on the same page: if you provide the snippet using the `otel.experimental.javascript-snippet` property, you can no longer override it via `setSnippet()` -- is that correct?
If so, can you add a comment that documents this behavior?

Reply:
That's correct, the user can only set the snippet once, regardless of the method they use. Thank you for the reminder, I will make sure to update the documentation to reflect this.